### ISSUE TITLE
[Bug/Support]: elpaca fails to do download dependency of parinfer-rust-mode

### ISSUE THREAD

**Issue by dmvianna** [progfolio/elpaca#311]
### Confirmation

- [X] I have checked the documentation (README, Wiki, docstrings, etc)
- [ ] I am  checking these without reading them.
- [X] I have searched previous issues to see if my question is a duplicate.

### Elpaca Version

Elpaca bc685f5 grafted, HEAD -> master, origin/master, origin/HEAD
installer:      0.7
emacs-version:  GNU Emacs 29.3 (build 1, x86_64-redhat-linux-gnu, GTK+ Version 3.24.41, cairo version 1.18.0)
 of 2024-04-26
git --version:  git version 2.45.1

### Operating System

Fedora 40

### Description

```lisp
(use-package parinfer-rust-mode
  :custom
  (parinfer-rust-troublesome-modes ;; removed electric-pair-mode from list
   '(hungry-delete-mode global-hungry-delete-mode))
  (parinfer-rust-auto-download t)
  (electric-pair-local-mode -1)
  (parinfer-rust-preferred-mode 'paren)
  :hook
  (emacs-lisp-mode
   ielm-mode
   lisp-mode
   slime-repl-mode
   lisp-interaction-mode
   racket-mode)
  (before-save . delete-trailing-whitespace))
```
I've been using this configuration for `parinfer-rust-mode` for about a year, but recently I ran `elpaca-update-all` and now I keep getting this error

```
Debugger entered--Lisp error: (file-missing "Cannot open load file" "No such file or directory" "track-changes")
  #<subr require>(track-changes)
  apply(#<subr require> track-changes)
  require(track-changes)
  load-with-code-conversion("/home/dmvianna/.config/emacs/elpaca/builds/parinfe..." "/home/dmvianna/.config/emacs/elpaca/builds/parinfe..." nil t)
  parinfer-rust-mode()
  run-hooks(change-major-mode-after-body-hook prog-mode-hook lisp-data-mode-hook emacs-lisp-mode-hook)
  apply(run-hooks (change-major-mode-after-body-hook prog-mode-hook lisp-data-mode-hook emacs-lisp-mode-hook))
  run-mode-hooks(emacs-lisp-mode-hook)
  emacs-lisp-mode()
  command-line()
  normal-top-level()
```
I have checked the `parinfer-rust-mode`'s source code and it does have `track-changes` listed as a [dependency](https://github.com/justinbarclay/parinfer-rust-mode/blob/e9a23e136b8c5f1fb718af93c19cd6ed22d8c98c/parinfer-rust-mode.el#L8).

So for the moment I'm assuming **elpaca** is not resolving the dependency as it should.

**Comment by progfolio** [comment]
Please run the following test and share the output here:

[How to run this test?](https://github.com/progfolio/elpaca/wiki/Troubleshooting#the-elpaca-test-macro)

```emacs-lisp
(elpaca-test
  :early-init (setq elpaca-menu-functions '(elpaca-menu-gnu-elpa))
  :init (elpaca (parinfer-rust-mode :repo "justinbarclay/parinfer-rust-mode" :host github :wait t))
  (princ (elpaca-info 'parinfer-rust-mode)))
```

**Comment by dmvianna** [comment]
<details open><summary>Test Case</summary>

[How to run this test?](https://github.com/progfolio/elpaca/wiki/Troubleshooting#the-elpaca-test-macro)

```emacs-lisp
(elpaca-test
  :early-init (setq elpaca-menu-functions '(elpaca-menu-gnu-elpa))
  :init
  (elpaca (parinfer-rust-mode :repo "justinbarclay/parinfer-rust-mode" :host github :wait t))
  (princ (elpaca-info 'parinfer-rust-mode)))
```

</details>
<details><summary>Host Env</summary>

<table>
<tr><td>elpaca</td><td>bc685f5 grafted, HEAD -> master, origin/master, origin/HEAD</td>
<tr><td>installer</td><td>0.7</td>
<tr><td>emacs</td><td>GNU Emacs 29.3 (build 1, x86_64-redhat-linux-gnu, GTK+ Version 3.24.41, cairo version 1.18.0) of 2024-04-26</td>
<tr><td>git</td><td>git version 2.45.1</td>
</table>
</details>

<details><summary>Output</summary>

```emacs-lisp
Elpaca menu item cache discarded due to version change.
  INFO     Scraping files for loaddefs... 
  INFO     Scraping files for loaddefs...done
  GEN      ../elpaca-autoloads.el
Cloning into '/tmp/elpaca.SVoulI/elpaca/repos/elpaca'...
Your branch is up to date with 'origin/master'.
Checking /tmp/elpaca.SVoulI/elpaca/repos/elpaca...
Compiling /tmp/elpaca.SVoulI/elpaca/repos/elpaca/elpaca-info.el...
Compiling /tmp/elpaca.SVoulI/elpaca/repos/elpaca/elpaca-log.el...
Compiling /tmp/elpaca.SVoulI/elpaca/repos/elpaca/elpaca-manager.el...
Compiling /tmp/elpaca.SVoulI/elpaca/repos/elpaca/elpaca-menu-elpa.el...
Compiling /tmp/elpaca.SVoulI/elpaca/repos/elpaca/elpaca-menu-melpa.el...
Compiling /tmp/elpaca.SVoulI/elpaca/repos/elpaca/elpaca-menu-org.el...
Compiling /tmp/elpaca.SVoulI/elpaca/repos/elpaca/elpaca-process.el...
Compiling /tmp/elpaca.SVoulI/elpaca/repos/elpaca/elpaca-test.el...
Compiling /tmp/elpaca.SVoulI/elpaca/repos/elpaca/elpaca-ui.el...
Compiling /tmp/elpaca.SVoulI/elpaca/repos/elpaca/elpaca.el...
Checking /tmp/elpaca.SVoulI/elpaca/repos/elpaca/doc...
Compiling /tmp/elpaca.SVoulI/elpaca/repos/elpaca/doc/early-init.el...
Compiling /tmp/elpaca.SVoulI/elpaca/repos/elpaca/doc/init.el...
Compiling /tmp/elpaca.SVoulI/elpaca/repos/elpaca/doc/installer.el...
Checking /tmp/elpaca.SVoulI/elpaca/repos/elpaca/extensions...
Compiling /tmp/elpaca.SVoulI/elpaca/repos/elpaca/extensions/elpaca-use-package.el...
Checking /tmp/elpaca.SVoulI/elpaca/repos/elpaca/images...
Checking /tmp/elpaca.SVoulI/elpaca/repos/elpaca/test...
Compiling /tmp/elpaca.SVoulI/elpaca/repos/elpaca/test/elpaca-test.el...
Compiling /tmp/elpaca.SVoulI/elpaca/repos/elpaca/test/elpaca-tests.el...
Done (Total of 12 files compiled, 4 skipped in 4 directories)

Downloading GNU ELPA...
parinfer-rust-mode
Not available in menu functions

source: Init file
full recipe:
( :package "parinfer-rust-mode"
  ;; Inherited from declaration.
  :wait t
  :host github
  :repo "justinbarclay/parinfer-rust-mode"
  ;; Inherited from elpaca-order-functions.
  :depth 1
  :inherit t
  :protocol https)
dependencies:
  emacs         >= 26.1
  track-changes >= 1.1
dependents: nil
installed version: 0.9.0 e9a23e1
statuses:
  (finished activation info byte-compilation autoloads linking unblocked blocked ref-checked-out cloning queued)
files:
  $REPOS/parinfer-rust-mode/parinfer-rust-changes.el → $BUILDS/parinfer-rust-mode/parinfer-rust-changes.el
  $REPOS/parinfer-rust-mode/parinfer-rust-helper.el  → $BUILDS/parinfer-rust-mode/parinfer-rust-helper.el
  $REPOS/parinfer-rust-mode/parinfer-rust-mode.el    → $BUILDS/parinfer-rust-mode/parinfer-rust-mode.el
log:
  [2024-06-03 12:38:56] Package queued
  [2024-06-03 12:38:56] Continued by: elpaca--process
  [2024-06-03 12:38:56] Cloning
  [2024-06-03 12:38:56] $git clone --depth 1 https://github.com/justinbarclay/parinfer-rust-mode.git /tmp/elpaca.SVoulI/elpaca/repos/parinfer-rust-mode/
  [2024-06-03 12:38:56]   Cloning into '/tmp/elpaca.SVoulI/elpaca/repos/parinfer-rust-mode'...
  [2024-06-03 12:38:57]   remote: Enumerating objects: 43, done.        
  [2024-06-03 12:38:57]   remote: Counting objects: 100% (43/43), done.        
  [2024-06-03 12:38:57]   remote: Compressing objects: 100% (36/36), done.        
  [2024-06-03 12:38:57]   remote: Total 43 (delta 4), reused 25 (delta 2), pack-reused 0        
  [2024-06-03 12:38:57]   Receiving objects: 100% (43/43), 2.26 MiB | 7.81 MiB/s, done.
  [2024-06-03 12:38:57]   Resolving deltas: 100% (4/4), done.
  [2024-06-03 12:38:57] Continued by: elpaca--clone-process-sentinel
  [2024-06-03 12:38:57] Continued by: elpaca--configure-remotes
  [2024-06-03 12:38:57] Continued by: elpaca--checkout-ref
  [2024-06-03 12:38:57] Continued by: elpaca--dispatch-build-commands
  [2024-06-03 12:38:57] Queueing Dependencies
  [2024-06-03 12:39:11] Continued by: elpaca--check-status
  [2024-06-03 12:39:11] Unblocked by: track-changes
  [2024-06-03 12:39:11] Checking dependency versions
  [2024-06-03 12:39:11] Continued by: elpaca--check-version
  [2024-06-03 12:39:11] Linking build files
  [2024-06-03 12:39:11] Continued by: elpaca--link-build-files
  [2024-06-03 12:39:11] Build files linked
  [2024-06-03 12:39:11] Generating autoloads: /tmp/elpaca.SVoulI/elpaca/builds/parinfer-rust-mode
  [2024-06-03 12:39:11] $/usr/bin/emacs -Q -L /tmp/elpaca.SVoulI/elpaca/repos/elpaca/ -l /tmp/elpaca.SVoulI/elpaca/repos/elpaca/elpaca.el --batch --eval (progn (setq gc-cons-percentage 1.0) (elpaca-generate-autoloads "parinfer-rust-mode" "/tmp/elpaca.SVoulI/elpaca/builds/parinfer-rust-mode"))
  [2024-06-03 12:39:12]     INFO     Scraping files for loaddefs... 
  [2024-06-03 12:39:12]     INFO     Scraping files for loaddefs...done
  [2024-06-03 12:39:12]     GEN      parinfer-rust-mode-autoloads.el
  [2024-06-03 12:39:12] Continued by: #[128 "\302\301\303\300\"\"\207" [("Autoloads Generated" nil) elpaca--process-sentinel apply append] 6 "

(fn &rest ARGS2)"]
  [2024-06-03 12:39:12] Autoloads Generated
  [2024-06-03 12:39:12] Byte compiling
  [2024-06-03 12:39:12] $/usr/bin/emacs -q --batch --eval (let ((gc-cons-percentage 1.0) (native-comp-eln-load-path '("/home/dmvianna/.config/emacs/eln-cache/" "/usr/lib64/emacs/29.3/native-lisp/"))) (dolist (dir '("/tmp/elpaca.SVoulI/elpaca/builds/parinfer-rust-mode" "/tmp/elpaca.SVoulI/elpaca/builds/track-changes")) (let ((default-directory dir)) (add-to-list 'load-path dir) (normal-top-level-add-subdirs-to-load-path))) (byte-recompile-directory "/tmp/elpaca.SVoulI/elpaca/builds/parinfer-rust-mode" 0 'force))
  [2024-06-03 12:39:12]   Loading /usr/share/emacs/site-lisp/site-start.d/autoconf-init.el (source)...
  [2024-06-03 12:39:12]   Loading /usr/share/emacs/site-lisp/site-start.d/desktop-entry-mode-init.el (source)...
  [2024-06-03 12:39:12]   Loading /usr/share/emacs/site-lisp/site-start.d/mercurial-site-start.el (source)...
  [2024-06-03 12:39:12]   Loading /usr/share/emacs/site-lisp/site-start.d/rpmdev-init.el (source)...
  [2024-06-03 12:39:12]   Checking /tmp/elpaca.SVoulI/elpaca/builds/parinfer-rust-mode...
  [2024-06-03 12:39:12]   Compiling /tmp/elpaca.SVoulI/elpaca/builds/parinfer-rust-mode/parinfer-rust-changes.el...
  [2024-06-03 12:39:12]   Compiling /tmp/elpaca.SVoulI/elpaca/builds/parinfer-rust-mode/parinfer-rust-helper.el...
  [2024-06-03 12:39:12]   Compiling /tmp/elpaca.SVoulI/elpaca/builds/parinfer-rust-mode/parinfer-rust-mode-autoloads.el...
  [2024-06-03 12:39:12]   Compiling /tmp/elpaca.SVoulI/elpaca/builds/parinfer-rust-mode/parinfer-rust-mode.el...
  [2024-06-03 12:39:12]   In toplevel form:
  [2024-06-03 12:39:12]   parinfer-rust-mode.el:455:2: Warning: defvar `parinfer-rust--in-debug' docstring wider than 80 characters
  [2024-06-03 12:39:12]   In parinfer-rust--execute-change-buffer-p:
  [2024-06-03 12:39:12]   parinfer-rust-mode.el:518:2: Warning: docstring wider than 80 characters
  [2024-06-03 12:39:12]   In parinfer-rust--set-options:
  [2024-06-03 12:39:12]   parinfer-rust-mode.el:533:2: Warning: docstring wider than 80 characters
  [2024-06-03 12:39:12]   Done (Total of 3 files compiled, 1 skipped)
  [2024-06-03 12:39:12] Continued by: #[128 "\302\301\303\300\"\"\207" [("Byte compilation complete" nil) elpaca--process-sentinel apply append] 6 "

(fn &rest ARGS2)"]
  [2024-06-03 12:39:12] Byte compilation complete
  [2024-06-03 12:39:12] Compiling Info files
  [2024-06-03 12:39:12] Continued by: elpaca--compile-info
  [2024-06-03 12:39:12] Info source files not found
  [2024-06-03 12:39:12] Continued by: elpaca--dispatch-build-commands
  [2024-06-03 12:39:12] Activating package
  [2024-06-03 12:39:12] Package build dir added to load-path
  [2024-06-03 12:39:12] Caching autoloads
  [2024-06-03 12:39:12] Autoloads cached
  [2024-06-03 12:39:12] Continued by: elpaca--activate-package
  [2024-06-03 12:39:12] ✓ 16.190 secs
```
</details>

**Comment by progfolio** [comment]
Thank you. The package builds cleanly in the test environment.

What's happened is parinfer-rust-mode added a new dependency since your last update.
Elpaca currently doesn't re-queue a package's dependencies when rebuilding after an update.
I'll look into extending it so it can detect such changes.

Have you restarted Emacs since updating the packages?
track-changes should be installed on demand in that case.
Please share the output of `M-x elpaca-log` and `M-x elpaca-info` `parinfer-rust-mode` after restarting Emacs.

**Comment by dmvianna** [comment]
```
 Package Status Info Time ▼ 
elpaca                         finished             ✓ 0.041 secs                                                                     00.044314
elpaca-use-package             finished             ✓ 0.041 secs                                                                     00.044675
use-package-ensure-system-package finished             ✓ 0.001 secs                                                                     00.058485
system-packages                finished             ✓ 0.000 secs                                                                     00.058487
with-editor                    finished             ✓ 0.000 secs                                                                     00.062422
compat                         finished             ✓ 0.001 secs                                                                     00.062425
magit-section                  finished             ✓ 0.003 secs                                                                     00.064626
dash                           finished             ✓ 0.003 secs                                                                     00.064630
magit                          finished             ✓ 0.010 secs                                                                     00.067996
git-commit                     finished             ✓ 0.005 secs                                                                     00.067997
gcmh                           finished             ✓ 1.391 secs                                                                     01.461871
auto-compile                   finished             ✓ 1.351 secs                                                                     01.462546
winum                          finished             ✓ 0.832 secs                                                                     01.463137
posframe                       finished             ✓ 0.000 secs                                                                     01.473974
cfrs                           finished             ✓ 0.002 secs                                                                     01.475180
s                              finished             ✓ 0.003 secs                                                                     01.475195
ht                             finished             ✓ 0.003 secs                                                                     01.475699
hydra                          finished             ✓ 0.005 secs                                                                     01.477592
lv                             finished             ✓ 0.000 secs                                                                     01.477597
pfuture                        finished             ✓ 0.006 secs                                                                     01.478153
treemacs                       finished             ✓ 0.814 secs                                                                     01.480797
ace-window                     finished             ✓ 0.008 secs                                                                     01.480818
avy                            finished             ✓ 0.603 secs                                                                     01.480824
solarized-theme                finished             ✓ 0.800 secs                                                                     01.481492
all-the-icons                  finished             ✓ 0.801 secs                                                                     01.482349
all-the-icons-dired            finished             ✓ 0.801 secs                                                                     01.482988
immortal-scratch               finished             ✓ 0.801 secs                                                                     01.483494
info-colors                    finished             ✓ 0.801 secs                                                                     01.484069
which-key                      finished             ✓ 0.802 secs                                                                     01.484937
delight                        finished             ✓ 0.802 secs                                                                     01.485489
string-inflection              finished             ✓ 0.803 secs                                                                     01.486085
visual-regexp                  finished             ✓ 0.803 secs                                                                     01.486644
visual-regexp-steroids         finished             ✓ 0.803 secs                                                                     01.487180
multiple-cursors               finished             ✓ 0.804 secs                                                                     01.488024
ibuffer-vc                     finished             ✓ 0.798 secs                                                                     01.488579
drag-stuff                     finished             ✓ 0.798 secs                                                                     01.489282
smart-mode-line                finished             ✓ 0.801 secs                                                                     01.492530
rich-minority                  finished             ✓ 0.001 secs                                                                     01.492542
dirvish                        finished             ✓ 0.805 secs                                                                     01.496311
transient                      finished             ✓ 0.799 secs                                                                     01.496317
eldoc-box                      finished             ✓ 0.805 secs                                                                     01.496843
rainbow-delimiters             finished             ✓ 0.805 secs                                                                     01.497436
highlight-indent-guides        finished             ✓ 0.806 secs                                                                     01.498262
flycheck                       finished             ✓ 0.808 secs                                                                     01.500327
aggressive-indent              finished             ✓ 0.808 secs                                                                     01.501011
language-id                    finished             ✓ 0.000 secs                                                                     01.502445
format-all                     finished             ✓ 0.810 secs                                                                     01.503429
inheritenv                     finished             ✓ 0.045 secs                                                                     01.503434
yasnippet                      finished             ✓ 0.812 secs                                                                     01.504925
origami                        finished             ✓ 0.812 secs                                                                     01.505590
puni                           finished             ✓ 0.813 secs                                                                     01.506614
combobulate                    finished             ✓ 0.813 secs                                                                     01.507339
undo-hl                        finished             ✓ 0.813 secs                                                                     01.507794
vundo                          finished             ✓ 0.813 secs                                                                     01.508378
rg                             finished             ✓ 0.815 secs                                                                     01.510308
wgrep                          finished             ✓ 0.000 secs                                                                     01.510314
dumb-jump                      finished             ✓ 0.817 secs                                                                     01.512287
popup                          finished             ✓ 0.000 secs                                                                     01.512293
exec-path-from-shell           finished             ✓ 0.817 secs                                                                     01.512755
multi-term                     finished             ✓ 0.817 secs                                                                     01.513487
aweshell                       finished             ✓ 0.818 secs                                                                     01.514372
abridge-diff                   finished             ✓ 0.815 secs                                                                     01.515017
consult-gh                     finished             ✓ 0.817 secs                                                                     01.517001
consult                        finished             ✓ 0.638 secs                                                                     01.517006
extmap                         finished             ✓ 0.000 secs                                                                     01.518890
logview                        finished             ✓ 0.800 secs                                                                     01.520133
datetime                       finished             ✓ 0.001 secs                                                                     01.520139
google-this                    finished             ✓ 0.800 secs                                                                     01.520645
sudo-edit                      finished             ✓ 0.800 secs                                                                     01.521219
markdown-mode                  finished             ✓ 0.657 secs                                                                     01.525281
spinner                        finished             ✓ 0.002 secs                                                                     01.526292
lsp-mode                       finished             ✓ 0.807 secs                                                                     01.528665
f                              finished             ✓ 0.004 secs                                                                     01.528676
lsp-ui                         finished             ✓ 0.808 secs                                                                     01.529444
lsp-treemacs                   finished             ✓ 0.808 secs                                                                     01.530240
polymode                       finished             ✓ 0.000 secs                                                                     01.535562
deferred                       finished             ✓ 0.001 secs                                                                     01.536090
request                        finished             ✓ 0.002 secs                                                                     01.536656
anaphora                       finished             ✓ 0.003 secs                                                                     01.537317
ein                            finished             ✓ 0.681 secs                                                                     01.538368
websocket                      finished             ✓ 0.004 secs                                                                     01.538374
csv-mode                       finished             ✓ 0.677 secs                                                                     01.539148
dhall-mode                     finished             ✓ 0.678 secs                                                                     01.540332
reformatter                    finished             ✓ 0.000 secs                                                                     01.540338
dockerfile-mode                finished             ✓ 0.678 secs                                                                     01.540965
elm-mode                       finished             ✓ 0.678 secs                                                                     01.541543
go-mode                        finished             ✓ 0.679 secs                                                                     01.542231
go-flymake                     finished             ✓ 0.679 secs                                                                     01.542590
graphviz-dot-mode              finished             ✓ 0.679 secs                                                                     01.543241
handlebars-mode                finished             ✓ 0.680 secs                                                                     01.543660
lsp-haskell                    finished             ✓ 0.682 secs                                                                     01.546705
haskell-mode                   finished             ✓ 0.001 secs                                                                     01.546711
json-mode                      finished             ✓ 0.683 secs                                                                     01.548433
json-snatcher                  finished             ✓ 0.000 secs                                                                     01.548440
lsp-docker                     finished             ✓ 0.001 secs                                                                     01.551416
yaml                           finished             ✓ 0.000 secs                                                                     01.551422
lsp-java                       finished             ✓ 0.688 secs                                                                     01.552786
dap-mode                       finished             ✓ 0.003 secs                                                                     01.552794
bui                            finished             ✓ 0.002 secs                                                                     01.552798
web-mode                       finished             ✓ 0.689 secs                                                                     01.554627
prettier-js                    finished             ✓ 0.689 secs                                                                     01.555258
typescript-mode                finished             ✓ 0.689 secs                                                                     01.555882
tree-sitter                    finished             ✓ 0.696 secs                                                                     01.562910
tsc                            finished             ✓ 0.002 secs                                                                     01.562918
tree-sitter-langs              finished             ✓ 0.696 secs                                                                     01.563432
tide                           finished             ✓ 0.697 secs                                                                     01.564195
pickle                         finished             ✓ 0.697 secs                                                                     01.564689
slime                          finished             ✓ 0.699 secs                                                                     01.566951
macrostep                      finished             ✓ 0.000 secs                                                                     01.566957
py-isort                       finished             ✓ 0.699 secs                                                                     01.567353
python-black                   finished             ✓ 0.699 secs                                                                     01.567822
lsp-jedi                       finished             ✓ 0.699 secs                                                                     01.568209
nix-mode                       finished             ✓ 0.699 secs                                                                     01.568943
racket-mode                    finished             ✓ 0.700 secs                                                                     01.569726
scala-mode                     finished             ✓ 0.700 secs                                                                     01.570784
sbt-mode                       finished             ✓ 0.701 secs                                                                     01.572010
lsp-metals                     finished             ✓ 0.701 secs                                                                     01.572665
theta-mode                     finished             ✓ 0.711 secs                                                                     01.582624
yaml-mode                      finished             ✓ 0.711 secs                                                                     01.583186
terraform-mode                 finished             ✓ 0.712 secs                                                                     01.584800
hcl-mode                       finished             ✓ 0.000 secs                                                                     01.584848
udev-mode                      finished             ✓ 0.713 secs                                                                     01.585407
hoon-mode                      finished             ✓ 0.713 secs                                                                     01.585927
hoon-ts-mode                   finished             ✓ 0.713 secs                                                                     01.586313
sql-indent                     finished             ✓ 0.714 secs                                                                     01.587119
bigquery-mode                  finished             ✓ 0.714 secs                                                                     01.587534
pdf-tools                      finished             ✓ 0.718 secs                                                                     01.592406
tablist                        finished             ✓ 0.000 secs                                                                     01.592412
nov                            finished             ✓ 0.719 secs                                                                     01.594565
esxml                          finished             ✓ 0.001 secs                                                                     01.594571
kv                             finished             ✓ 0.000 secs                                                                     01.594575
vertico                        finished             ✓ 0.719 secs                                                                     01.595404
marginalia                     finished             ✓ 0.719 secs                                                                     01.596003
corfu                          finished             ✓ 0.720 secs                                                                     01.596669
cape                           finished             ✓ 0.721 secs                                                                     01.597597
corfu-popupinfo                finished             ✓ 0.721 secs                                                                     01.598405
corfu-history                  finished             ✓ 0.722 secs                                                                     01.599150
corfu-terminal                 finished             ✓ 0.723 secs                                                                     01.600677
popon                          finished             ✓ 0.000 secs                                                                     01.600683
kind-icon                      finished             ✓ 0.724 secs                                                                     01.601891
svg-lib                        finished             ✓ 0.000 secs                                                                     01.601896
embark                         finished             ✓ 0.724 secs                                                                     01.602729
embark-consult                 finished             ✓ 0.724 secs                                                                     01.603198
consult-eglot                  finished             ✓ 0.725 secs                                                                     01.603608
orderless                      finished             ✓ 0.725 secs                                                                     01.604179
emms                           finished             ✓ 0.725 secs                                                                     01.605274
emojify                        finished             ✓ 0.000 secs                                                                     01.606695
log4e                          finished             ✓ 0.000 secs                                                                     01.607933
alert                          finished             ✓ 0.491 secs                                                                     01.608843
gntp                           finished             ✓ 0.001 secs                                                                     01.608850
circe                          finished             ✓ 0.003 secs                                                                     01.609562
slack                          finished             ✓ 0.494 secs                                                                     01.610631
oauth2                         finished             ✓ 0.004 secs                                                                     01.610636
mastodon                       finished             ✓ 0.494 secs                                                                     01.613285
persist                        finished             ✓ 0.000 secs                                                                     01.613293
org                            finished             ✓ 0.503 secs                                                                     01.623867
org-pandoc-import              finished             ✓ 0.502 secs                                                                     01.624455
org-roam                       finished             ✓ 0.503 secs                                                                     01.626390
emacsql                        finished             ✓ 0.000 secs                                                                     01.626396
org-journal                    finished             ✓ 0.502 secs                                                                     01.627153
org-download                   finished             ✓ 0.502 secs                                                                     01.628907
async                          finished             ✓ 0.000 secs                                                                     01.628915
verb                           finished             ✓ 0.502 secs                                                                     01.629620
simple-httpd                   finished             ✓ 0.000 secs                                                                     01.630902
jupyter                        finished             ✓ 0.503 secs                                                                     01.632177
zmq                            finished             ✓ 0.001 secs                                                                     01.632184
envrc                          finished             ✓ 0.174 secs                                                                     01.632714
```

**Comment by dmvianna** [comment]
```
parinfer-rust-mode
An interface for the parinfer-rust library

source: MELPA
url: https://github.com/justinbarclay/parinfer-rust-mode
menu item recipe:
( :package "parinfer-rust-mode"
  :repo "justinbarclay/parinfer-rust-mode"
  :fetcher github
  :files ("*.el" "*.el.in" "dir" "*.info"
          "*.texi" "*.texinfo" "doc/dir"
          "doc/*.info" "doc/*.texi"
          "doc/*.texinfo" "lisp/*.el" (:exclude
                                       ".dir-locals.el" "test.el" "tests.el"
                                       "*-test.el" "*-tests.el" "LICENSE"
                                       "README*" "*-pkg.el")))
full recipe:
( :package "parinfer-rust-mode"
  ;; Inherited from elpaca-order-functions.
  :depth 1
  :inherit t
  :protocol https
  ;; Inherited from elpaca-menu-item.
  :files (:defaults)
  :fetcher github
  :repo "justinbarclay/parinfer-rust-mode")
dependencies:
?
dependents: ?
```

**Comment by dmvianna** [comment]
```
 Package Status Info Time ▼ 
elpaca                         finished             ✓ 0.038 secs                                                                     00.040911
elpaca-use-package             finished             ✓ 0.038 secs                                                                     00.041176
use-package-ensure-system-package finished             ✓ 0.001 secs                                                                     00.053034
system-packages                finished             ✓ 0.000 secs                                                                     00.053035
with-editor                    finished             ✓ 0.000 secs                                                                     00.055646
compat                         finished             ✓ 0.001 secs                                                                     00.055649
magit-section                  finished             ✓ 0.002 secs                                                                     00.057691
dash                           finished             ✓ 0.003 secs                                                                     00.057694
magit                          finished             ✓ 0.008 secs                                                                     00.060849
git-commit                     finished             ✓ 0.004 secs                                                                     00.060851
gcmh                           finished             ✓ 1.347 secs                                                                     01.410659
auto-compile                   finished             ✓ 1.315 secs                                                                     01.413234
winum                          finished             ✓ 0.784 secs                                                                     01.415687
posframe                       finished             ✓ 0.001 secs                                                                     01.440151
cfrs                           finished             ✓ 0.003 secs                                                                     01.441459
s                              finished             ✓ 0.004 secs                                                                     01.441485
ht                             finished             ✓ 0.004 secs                                                                     01.442128
hydra                          finished             ✓ 0.006 secs                                                                     01.444147
lv                             finished             ✓ 0.000 secs                                                                     01.444153
pfuture                        finished             ✓ 0.007 secs                                                                     01.444865
treemacs                       finished             ✓ 0.774 secs                                                                     01.447858
ace-window                     finished             ✓ 0.009 secs                                                                     01.447865
avy                            finished             ✓ 0.567 secs                                                                     01.447870
solarized-theme                finished             ✓ 0.760 secs                                                                     01.448627
all-the-icons                  finished             ✓ 0.761 secs                                                                     01.449560
all-the-icons-dired            finished             ✓ 0.761 secs                                                                     01.450148
immortal-scratch               finished             ✓ 0.761 secs                                                                     01.450744
info-colors                    finished             ✓ 0.761 secs                                                                     01.451248
which-key                      finished             ✓ 0.762 secs                                                                     01.452193
delight                        finished             ✓ 0.763 secs                                                                     01.452776
string-inflection              finished             ✓ 0.763 secs                                                                     01.453383
visual-regexp                  finished             ✓ 0.763 secs                                                                     01.453942
visual-regexp-steroids         finished             ✓ 0.764 secs                                                                     01.454478
multiple-cursors               finished             ✓ 0.764 secs                                                                     01.455309
ibuffer-vc                     finished             ✓ 0.759 secs                                                                     01.455809
drag-stuff                     finished             ✓ 0.760 secs                                                                     01.456356
smart-mode-line                finished             ✓ 0.761 secs                                                                     01.457754
rich-minority                  finished             ✓ 0.000 secs                                                                     01.457760
dirvish                        finished             ✓ 0.763 secs                                                                     01.459802
transient                      finished             ✓ 0.758 secs                                                                     01.459808
eldoc-box                      finished             ✓ 0.763 secs                                                                     01.460356
rainbow-delimiters             finished             ✓ 0.763 secs                                                                     01.460854
highlight-indent-guides        finished             ✓ 0.763 secs                                                                     01.461370
flycheck                       finished             ✓ 0.765 secs                                                                     01.463118
aggressive-indent              finished             ✓ 0.765 secs                                                                     01.463669
language-id                    finished             ✓ 0.000 secs                                                                     01.464461
format-all                     finished             ✓ 0.766 secs                                                                     01.465120
inheritenv                     finished             ✓ 0.068 secs                                                                     01.465153
yasnippet                      finished             ✓ 0.767 secs                                                                     01.466046
origami                        finished             ✓ 0.767 secs                                                                     01.466547
puni                           finished             ✓ 0.768 secs                                                                     01.467221
combobulate                    finished             ✓ 0.768 secs                                                                     01.467777
undo-hl                        finished             ✓ 0.768 secs                                                                     01.468148
vundo                          finished             ✓ 0.769 secs                                                                     01.468701
rg                             finished             ✓ 0.770 secs                                                                     01.470034
wgrep                          finished             ✓ 0.000 secs                                                                     01.470039
dumb-jump                      finished             ✓ 0.771 secs                                                                     01.471271
popup                          finished             ✓ 0.000 secs                                                                     01.471277
exec-path-from-shell           finished             ✓ 0.771 secs                                                                     01.471736
multi-term                     finished             ✓ 0.771 secs                                                                     01.472270
aweshell                       finished             ✓ 0.772 secs                                                                     01.472884
abridge-diff                   finished             ✓ 0.769 secs                                                                     01.473381
consult-gh                     finished             ✓ 0.770 secs                                                                     01.475149
consult                        finished             ✓ 0.593 secs                                                                     01.475154
extmap                         finished             ✓ 0.000 secs                                                                     01.476550
logview                        finished             ✓ 0.753 secs                                                                     01.477330
datetime                       finished             ✓ 0.001 secs                                                                     01.477336
google-this                    finished             ✓ 0.752 secs                                                                     01.477778
sudo-edit                      finished             ✓ 0.752 secs                                                                     01.478215
markdown-mode                  finished             ✓ 0.609 secs                                                                     01.480396
spinner                        finished             ✓ 0.001 secs                                                                     01.480888
lsp-mode                       finished             ✓ 0.756 secs                                                                     01.482095
f                              finished             ✓ 0.002 secs                                                                     01.482101
lsp-ui                         finished             ✓ 0.756 secs                                                                     01.482540
lsp-treemacs                   finished             ✓ 0.756 secs                                                                     01.483119
polymode                       finished             ✓ 0.000 secs                                                                     01.486409
deferred                       finished             ✓ 0.001 secs                                                                     01.486964
request                        finished             ✓ 0.002 secs                                                                     01.487510
anaphora                       finished             ✓ 0.002 secs                                                                     01.488065
ein                            finished             ✓ 0.629 secs                                                                     01.488991
websocket                      finished             ✓ 0.003 secs                                                                     01.488997
csv-mode                       finished             ✓ 0.624 secs                                                                     01.489624
dhall-mode                     finished             ✓ 0.625 secs                                                                     01.490607
reformatter                    finished             ✓ 0.000 secs                                                                     01.490612
dockerfile-mode                finished             ✓ 0.625 secs                                                                     01.491024
elm-mode                       finished             ✓ 0.625 secs                                                                     01.491532
go-mode                        finished             ✓ 0.626 secs                                                                     01.492088
go-flymake                     finished             ✓ 0.626 secs                                                                     01.492425
graphviz-dot-mode              finished             ✓ 0.626 secs                                                                     01.492956
handlebars-mode                finished             ✓ 0.626 secs                                                                     01.493348
lsp-haskell                    finished             ✓ 0.628 secs                                                                     01.495013
haskell-mode                   finished             ✓ 0.001 secs                                                                     01.495019
json-mode                      finished             ✓ 0.628 secs                                                                     01.496055
json-snatcher                  finished             ✓ 0.000 secs                                                                     01.496062
lsp-docker                     finished             ✓ 0.001 secs                                                                     01.498466
yaml                           finished             ✓ 0.000 secs                                                                     01.498470
lsp-java                       finished             ✓ 0.631 secs                                                                     01.499664
dap-mode                       finished             ✓ 0.002 secs                                                                     01.499670
bui                            finished             ✓ 0.001 secs                                                                     01.499674
web-mode                       finished             ✓ 0.632 secs                                                                     01.501222
prettier-js                    finished             ✓ 0.632 secs                                                                     01.501612
typescript-mode                finished             ✓ 0.633 secs                                                                     01.502149
tree-sitter                    finished             ✓ 0.638 secs                                                                     01.507660
tsc                            finished             ✓ 0.002 secs                                                                     01.507667
tree-sitter-langs              finished             ✓ 0.638 secs                                                                     01.508348
tide                           finished             ✓ 0.639 secs                                                                     01.509027
pickle                         finished             ✓ 0.639 secs                                                                     01.509413
slime                          finished             ✓ 0.640 secs                                                                     01.511321
macrostep                      finished             ✓ 0.000 secs                                                                     01.511327
parinfer-rust-mode             finished             ✓ 0.654 secs                                                                     01.525299
track-changes                  finished             ✓ 0.001 secs                                                                     01.525306
py-isort                       finished             ✓ 0.654 secs                                                                     01.525782
python-black                   finished             ✓ 0.654 secs                                                                     01.526248
lsp-jedi                       finished             ✓ 0.654 secs                                                                     01.526623
nix-mode                       finished             ✓ 0.654 secs                                                                     01.527346
racket-mode                    finished             ✓ 0.655 secs                                                                     01.528092
scala-mode                     finished             ✓ 0.654 secs                                                                     01.528582
sbt-mode                       finished             ✓ 0.654 secs                                                                     01.529215
lsp-metals                     finished             ✓ 0.655 secs                                                                     01.529745
theta-mode                     finished             ✓ 0.661 secs                                                                     01.536552
yaml-mode                      finished             ✓ 0.661 secs                                                                     01.536980
terraform-mode                 finished             ✓ 0.662 secs                                                                     01.538339
hcl-mode                       finished             ✓ 0.000 secs                                                                     01.538345
udev-mode                      finished             ✓ 0.662 secs                                                                     01.538706
hoon-mode                      finished             ✓ 0.662 secs                                                                     01.539141
hoon-ts-mode                   finished             ✓ 0.662 secs                                                                     01.539508
sql-indent                     finished             ✓ 0.663 secs                                                                     01.540162
bigquery-mode                  finished             ✓ 0.663 secs                                                                     01.540549
pdf-tools                      finished             ✓ 0.666 secs                                                                     01.544225
tablist                        finished             ✓ 0.000 secs                                                                     01.544232
nov                            finished             ✓ 0.667 secs                                                                     01.545832
esxml                          finished             ✓ 0.000 secs                                                                     01.545837
kv                             finished             ✓ 0.000 secs                                                                     01.545840
vertico                        finished             ✓ 0.667 secs                                                                     01.546565
marginalia                     finished             ✓ 0.667 secs                                                                     01.547070
corfu                          finished             ✓ 0.667 secs                                                                     01.547712
cape                           finished             ✓ 0.668 secs                                                                     01.548336
corfu-popupinfo                finished             ✓ 0.668 secs                                                                     01.549015
corfu-history                  finished             ✓ 0.669 secs                                                                     01.549674
corfu-terminal                 finished             ✓ 0.670 secs                                                                     01.550923
popon                          finished             ✓ 0.000 secs                                                                     01.550929
kind-icon                      finished             ✓ 0.670 secs                                                                     01.552011
svg-lib                        finished             ✓ 0.000 secs                                                                     01.552020
embark                         finished             ✓ 0.670 secs                                                                     01.552862
embark-consult                 finished             ✓ 0.671 secs                                                                     01.553251
consult-eglot                  finished             ✓ 0.671 secs                                                                     01.553649
orderless                      finished             ✓ 0.671 secs                                                                     01.554139
emms                           finished             ✓ 0.671 secs                                                                     01.555025
emojify                        finished             ✓ 0.000 secs                                                                     01.556376
log4e                          finished             ✓ 0.000 secs                                                                     01.557448
alert                          finished             ✓ 0.439 secs                                                                     01.558156
gntp                           finished             ✓ 0.001 secs                                                                     01.558161
circe                          finished             ✓ 0.003 secs                                                                     01.558795
slack                          finished             ✓ 0.441 secs                                                                     01.559904
oauth2                         finished             ✓ 0.003 secs                                                                     01.559911
mastodon                       finished             ✓ 0.443 secs                                                                     01.562079
persist                        finished             ✓ 0.000 secs                                                                     01.562086
org                            finished             ✓ 0.456 secs                                                                     01.575766
org-pandoc-import              finished             ✓ 0.456 secs                                                                     01.576371
org-roam                       finished             ✓ 0.458 secs                                                                     01.578334
emacsql                        finished             ✓ 0.000 secs                                                                     01.578343
org-journal                    finished             ✓ 0.458 secs                                                                     01.578948
org-download                   finished             ✓ 0.460 secs                                                                     01.580627
async                          finished             ✓ 0.000 secs                                                                     01.580634
verb                           finished             ✓ 0.460 secs                                                                     01.581195
simple-httpd                   finished             ✓ 0.000 secs                                                                     01.582521
jupyter                        finished             ✓ 0.462 secs                                                                     01.583714
zmq                            finished             ✓ 0.001 secs                                                                     01.583720
envrc                          finished             ✓ 0.186 secs                                                                     01.584229
```

**Comment by dmvianna** [comment]
```
parinfer-rust-mode
An interface for the parinfer-rust library

source: MELPA
url: https://github.com/justinbarclay/parinfer-rust-mode
menu item recipe:
( :package "parinfer-rust-mode"
  :repo "justinbarclay/parinfer-rust-mode"
  :fetcher github
  :files ("*.el" "*.el.in" "dir" "*.info"
          "*.texi" "*.texinfo" "doc/dir"
          "doc/*.info" "doc/*.texi"
          "doc/*.texinfo" "lisp/*.el" (:exclude
                                       ".dir-locals.el" "test.el" "tests.el"
                                       "*-test.el" "*-tests.el" "LICENSE"
                                       "README*" "*-pkg.el"))
  :source "MELPA")
full recipe:
( :package "parinfer-rust-mode"
  ;; Inherited from elpaca-order-functions.
  :depth 1
  :inherit t
  :protocol https
  ;; Inherited from elpaca-menu-item.
  :source "MELPA"
  :files (:defaults)
  :fetcher github
  :repo "justinbarclay/parinfer-rust-mode")
dependencies:
  emacs         >= 26.1
  track-changes >= 1.1
dependents: nil
installed version: 0.9.0 e9a23e1
statuses:
  (finished activation info unblocked blocked queued)
files:
  $REPOS/parinfer-rust-mode/parinfer-rust-changes.el → $BUILDS/parinfer-rust-mode/parinfer-rust-changes.el
  $REPOS/parinfer-rust-mode/parinfer-rust-helper.el  → $BUILDS/parinfer-rust-mode/parinfer-rust-helper.el
  $REPOS/parinfer-rust-mode/parinfer-rust-mode.el    → $BUILDS/parinfer-rust-mode/parinfer-rust-mode.el
log:
  [2024-06-04 12:30:06] Package queued
  [2024-06-04 12:30:07] Continued by: elpaca--process
  [2024-06-04 12:30:07] Queueing Dependencies
  [2024-06-04 12:30:07] Continued by: elpaca--check-status
  [2024-06-04 12:30:07] Unblocked by: track-changes
  [2024-06-04 12:30:07] No Info dir file found
  [2024-06-04 12:30:07] Continued by: elpaca--add-info-path
  [2024-06-04 12:30:07] Activating package
  [2024-06-04 12:30:07] Package build dir added to load-path
  [2024-06-04 12:30:07] Caching autoloads
  [2024-06-04 12:30:07] Autoloads cached
  [2024-06-04 12:30:07] Continued by: elpaca--activate-package
  [2024-06-04 12:30:07] ✓ 0.654 secs
```

**Comment by dmvianna** [comment]
I had commended out `parinfer-rust-mode` from my config, so I had to delete my initial comments and replace them with the results of adding `parinfer-rust-mode` to the config again.

**Comment by dmvianna** [comment]
So **elpaca** does not complain, but I do get this in my `*Messages*` buffer when starting:
```
custom-initialize-reset: Cannot open load file: No such file or directory, track-changes
```

**Comment by progfolio** [comment]
Sorry. Somehow edited your last comment instead of posting my own...

Sounds like maybe a a form saved by Emacs customization system is responsible.
If you have a custom.el file, I'd check it to see if there's anything related to track-changes in there.

**Comment by dmvianna** [comment]
I do have `custom.el`, but a search on it doesn't give me any `track-changes` results. The closest thing is `tramp...`.

**Comment by progfolio** [comment]
> I do have `custom.el`, but a search on it doesn't give me any `track-changes` results. The closest thing is `tramp...`.

Try moving the loading of that file to `elpaca-after-init-hook`.

**Comment by dmvianna** [comment]
I deleted the `custom.el` and removed it from my config. `emacs --debug-init` still complains of the missing `track-changes`.

**Comment by dmvianna** [comment]
Interesting. I renamed the `elpaca` directory, and `emacs --debug-init` fails on something related to its cache.

**Comment by dmvianna** [comment]
`*messages*`
```
Loading /usr/share/emacs/site-lisp/site-start.d/autoconf-init.el (source)...done
Loading /usr/share/emacs/site-lisp/site-start.d/desktop-entry-mode-init.el (source)...done
Loading /usr/share/emacs/site-lisp/site-start.d/mercurial-site-start.el (source)...done
Loading /usr/share/emacs/site-lisp/site-start.d/rpmdev-init.el (source)...done
Elpaca menu item cache discarded due to version change.
  INFO     Scraping files for loaddefs...done
  GEN      ../elpaca-autoloads.el
Cloning into '/home/dmvianna/.config/emacs/elpaca/repos/elpaca'...
Your branch is up to date with 'origin/master'.
Checking /home/dmvianna/.config/emacs/elpaca/repos/elpaca...
Compiling /home/dmvianna/.config/emacs/elpaca/repos/elpaca/elpaca-info.el...
Elpaca menu item cache discarded due to version change.
Compiling /home/dmvianna/.config/emacs/elpaca/repos/elpaca/elpaca-log.el...
Compiling /home/dmvianna/.config/emacs/elpaca/repos/elpaca/elpaca-manager.el...
Compiling /home/dmvianna/.config/emacs/elpaca/repos/elpaca/elpaca-menu-elpa.el...
Compiling /home/dmvianna/.config/emacs/elpaca/repos/elpaca/elpaca-menu-melpa.el...
Compiling /home/dmvianna/.config/emacs/elpaca/repos/elpaca/elpaca-menu-org.el...
Compiling /home/dmvianna/.config/emacs/elpaca/repos/elpaca/elpaca-process.el...
Compiling /home/dmvianna/.config/emacs/elpaca/repos/elpaca/elpaca-test.el...
Compiling /home/dmvianna/.config/emacs/elpaca/repos/elpaca/elpaca-ui.el...
Compiling /home/dmvianna/.config/emacs/elpaca/repos/elpaca/elpaca.el...
Checking /home/dmvianna/.config/emacs/elpaca/repos/elpaca/doc...
Compiling /home/dmvianna/.config/emacs/elpaca/repos/elpaca/doc/early-init.el...
Compiling /home/dmvianna/.config/emacs/elpaca/repos/elpaca/doc/init.el...
Compiling /home/dmvianna/.config/emacs/elpaca/repos/elpaca/doc/installer.el...
Checking /home/dmvianna/.config/emacs/elpaca/repos/elpaca/extensions...
Compiling /home/dmvianna/.config/emacs/elpaca/repos/elpaca/extensions/elpaca-use-package.el...
Checking /home/dmvianna/.config/emacs/elpaca/repos/elpaca/images...
Checking /home/dmvianna/.config/emacs/elpaca/repos/elpaca/test...
Compiling /home/dmvianna/.config/emacs/elpaca/repos/elpaca/test/elpaca-test.el...
Compiling /home/dmvianna/.config/emacs/elpaca/repos/elpaca/test/elpaca-tests.el...
Done (Total of 12 files compiled, 4 skipped in 4 directories)

Downloading MELPA recipes...
Added to /home/dmvianna/.config/emacs/elpaca/cache/melpa/.git/info/sparse-checkout
Downloading MELPA recipes...100%
Downloading NonGNU-devel ELPA...
Downloading GNU-devel ELPA...
Entering debugger...
Beginning of buffer [2 times]
```
`*backtrace*`
```
Debugger entered--Lisp error: (wrong-type-argument stringp nil)
  elpaca-menu-elpa--metadata(((cache) (name . "NonGNU ELPA") (cache-path . "/home/dmvianna/.config/emacs/elpaca/cache/non-gnu-...") (packages-url . "https://git.savannah.gnu.org/cgit/emacs/nongnu.git...") (metadata-url . "https://elpa.nongnu.org/nongnu/") (remote . "git://git.sv.gnu.org/emacs/nongnu") (branch-prefix . "elpa")))
  elpaca-menu-elpa--index(((cache) (name . "NonGNU ELPA") (cache-path . "/home/dmvianna/.config/emacs/elpaca/cache/non-gnu-...") (packages-url . "https://git.savannah.gnu.org/cgit/emacs/nongnu.git...") (metadata-url . "https://elpa.nongnu.org/nongnu/") (remote . "git://git.sv.gnu.org/emacs/nongnu") (branch-prefix . "elpa")))
  elpaca-menu--elpa(((cache) (name . "NonGNU ELPA") (cache-path . "/home/dmvianna/.config/emacs/elpaca/cache/non-gnu-...") (packages-url . "https://git.savannah.gnu.org/cgit/emacs/nongnu.git...") (metadata-url . "https://elpa.nongnu.org/nongnu/") (remote . "git://git.sv.gnu.org/emacs/nongnu") (branch-prefix . "elpa")) index)
  elpaca-menu-non-gnu-elpa(index)
  elpaca--menu-items(t)
  elpaca-menu-item(elpaca nil)
  elpaca-recipe((elpaca :repo "https://github.com/progfolio/elpaca.git" :ref nil :depth 1 :files (:defaults "elpaca-test.el" (:exclude "extensions")) :build (:not elpaca--activate-package)))
  elpaca<-create((elpaca :repo "https://github.com/progfolio/elpaca.git" :ref nil :depth 1 :files (:defaults "elpaca-test.el" (:exclude "extensions")) :build (:not elpaca--activate-package)))
  elpaca--queue((elpaca :repo "https://github.com/progfolio/elpaca.git" :ref nil :depth 1 :files (:defaults "elpaca-test.el" (:exclude "extensions")) :build (:not elpaca--activate-package)) (elpaca-q init 0 0 incomplete (26207 41475 261424 871000) nil nil nil))
  elpaca--expand-declaration(`(,@elpaca-order) nil)
  load-with-code-conversion("/home/dmvianna/.config/emacs/init.el" "/home/dmvianna/.config/emacs/init.el" t t)
  load("/home/dmvianna/.config/emacs/init" noerror nomessage)
  #<subr startup--load-user-init-file>(#f(compiled-function () #<bytecode 0x1a680cb7f77a502a>) #f(compiled-function () #<bytecode 0x4d131ce0e73c34b>) t)
  apply(#<subr startup--load-user-init-file> (#f(compiled-function () #<bytecode 0x1a680cb7f77a502a>) #f(compiled-function () #<bytecode 0x4d131ce0e73c34b>) t))
  startup--load-user-init-file(#f(compiled-function () #<bytecode 0x1a680cb7f77a502a>) #f(compiled-function () #<bytecode 0x4d131ce0e73c34b>) t)
  command-line()
  normal-top-level()
```

**Comment by progfolio** [comment]
Do you have your configuration uploaded somewhere? I can take a look and see if I can reproduce the issue on my end.

**Comment by dmvianna** [comment]
OK, I fixed the problem. Once I deleted `~/.config/emacs/.cache` and `~/.config/emacs/.eln-cache` `parinfer-rust-mode` installed just fine.

You could not reproduce my issue exactly because your test environment is clean. Your test environment does not emulate upgrades. There's state being kept in the caches that it does not emulate successfully.

> What specifically did you do?
I renamed `~/.config/emacs/elpaca` to `~/.config/emacs/elpaca-off`, to test the hypothesis that the problem came from that directory's state. It didn't fix it, but it helped (maybe fortuitously) to point in the right direction.

Happy to close the issue. Thanks for your time and interest!

**Comment by progfolio** [comment]
> OK, I fixed the problem. Once I deleted `~/.config/emacs/.cache` and `~/.config/emacs/.eln-cache` `parinfer-rust-mode` installed just fine.
> 
> You could not reproduce my issue exactly because your test environment is clean. Your test environment does not emulate upgrades. There's state being kept in the caches that it does not emulate successfully.
> 
> > What specifically did you do?
> > I renamed `~/.config/emacs/elpaca` to `~/.config/emacs/elpaca-off`, to test the hypothesis that the problem came from that directory's state. It didn't fix it, but it helped (maybe fortuitously) to point in the right direction.
> 
> Happy to close the issue. Thanks for your time and interest!

Thanks for the details. Good to know in case anyone else runs into something similar. Glad you got things working again.
